docs/en/creating-mfa-method-frontend.md

```diff
@@ -0,0 +1,218 @@
+# Creating a new MFA method: Frontend
+
+## Introduction
+
+The MFA module provides a clear path for creating additional authentication methods. In this document we'll cover how to
+implement the front-end portion of the required code, using the Basic Math method as an example. Some prior experience
+with React / Redux is recommended.
+
+The front-end components of MFA make use of `react-injector` (Injector) to allow sharing of React components and Redux
+reducers between separate JS bundles. You can find more documentation on the Injector API in the [SilverStripe docs](https://docs.silverstripe.org/en/4/developer_guides/customising_the_admin_interface/reactjs_redux_and_graphql/#the-injector-api).
+
+You'll find it easiest to get up and running by matching the NPM dependencies and Webpack configuration used in the TOTP
+and WebAuthn modules, with a single entry point that handles registering your components with Injector.
+
+## Create components
+
+In order to handle both registration of your method, and authentication via it, you'll need to provide a component for
+each. The Register and Verify components in the core MFA module are designed to fetch and render your component when the
+user selects your method, either in the registration flow or when authenticating.
+
+### Register
+
+Your component for registration will need to accept a couple of key props:
+
+- `onCompleteRegistration`: A callback that should be invoked when your registration process is complete. Pass in an
+  object with any data that needs to be passed to your `RegisterHandlerInterface::register()` implementation to complete
+  the registration process.
+- `onBack`: A callback that should be invoked if the user wants to pick another method. We recommend rendering a 'Back'
+  button in the same fashion as the TOTP / WebAuthn methods do.
+- Any data you return from your `RegisterHandlerInterface::start()` implementation will also be provided to the
+  component as props. For example, the TOTP module sends a code to expose in the UI for the user to scan as a QR code or
+  enter manually into their authenticator app.
+
+A Register component for Basic Math might look like this:
+
+```jsx
+import React, { Component } from 'react';
+
+class BasicMathRegister extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      secret: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ secret: event.target.value });
+  }
+
+  render() {
+    const { onCompleteRegistration, onBack } = this.props;
+    const { ss: { i18n } } = window;
```

Comment:
Do you want to remove the internationalisation for simplicity?

Reply:
Nah, a good example is a complete example :)

Reply:
Since devs will be able to make use of the i18n module, I feel comfortable leaving it in place as a recommended approach, but @NightJar is right in that it should be documented. I'll add a note about it for now, we can expand on it later if it seems useful.